M1_MAC_SETUP.md

```diff
@@ -0,0 +1,92 @@
+parallel Homebrew versions, one for ARM and one for x86 (in Rosetta), create
+aliases for the two, and install all of `bigquery-etl`'s dependencies in
+Rosetta mode.
+
+For tips on maintaining parallel stacks of python and homebrew running with and without Rosetta, see blog posts from [Thinknum](https://medium.com/thinknum/how-to-install-python-under-rosetta-2-f98c0865e012) and [Sixty North](http://sixty-north.com/blog/pyenv-apple-silicon.html).
+
+1. Install Rosetta if you don't have it yet
+
+   ```zsh
+   softwareupdate --install-rosetta
+   ```
+
+2. Set Terminal to run in Rosetta mode:
+   1. In Finder, go to Applications -> Utilities
+   2. Right-click on Terminal and choose Get Info
+   3. Check the box "Open using Rosetta"
+   4. Open a new Terminal session and run `arch`. This will show you the
+      current architecture -- it should output `i386`. If you're not in Rosetta
+      mode, it will output `arm64`
+   5. (Optional) You can add the current architecture to your prompt -- add the
+      following to your `~/.zshrc` or `/.bashrc`:
+
+      ```zsh
+      setopt PROMPT_SUBST
+      PROMPT='$(arch) %~ %# '
+      ```
+
+3. Install Homebrew:
+   1. Still in your Rosetta mode shell, follow the instructions at
+      <https://brew.sh/> to install Homebrew
+   2. This will have installed Homebrew to `/usr/local/bin/brew`; now we'll add
+      some aliases -- add the following to your `~/.zshrc` or `/.bashrc`:
+
+      ```zsh
+      alias brew86='/usr/local/bin/brew'
+      alias brewm1='/opt/homebrew/bin/brew'
+      alias brew='echo "Use brew86 or brewm1"'
+      ```
+
+      You'll now use `brew86` to install things to Rosetta Homebrew and
+      `brewm1` to install to ARM Homebrew. The third alias I added because I
+      kept just typing `brew` out of habit without knowing which one I was
+      using
+   3. Delete the following line from your `~/.zprofile` or `~/bash_profile` if
+      you previously added it:
+
+      ```zsh
+      eval "$(/opt/homebrew/bin/brew shellenv)"
+      ```
+
+4. (Recommended) Clear out non-Rosetta versions of software that's used in
+   `bigquery-etl`:
+   1. If you installed Python 3.10 via pyenv, uninstall it with
+
+      ```zsh
+      pyenv uninstall 3.10.6  # (or whichever version you have installed)
+      ```
+
+   2. If you installed pyenv, Java, or Maven already, uninstall them (this is
+      the Java setup I have; yours might be different):
+
+      ```zsh
+      brewm1 uninstall pyenv openjdk maven temurin
+      ```
+
+   3. If you're using tmux or something like that, uninstall that too and
+      reinstall via `brew86`
+5. Install the core `bigquery-etl` dependencies via `brew86`:
+
+   ```zsh
+   brew86 install pyenv openjdk maven
+   brew86 install --cask temurin
+   ```
+
+6. Install python; you'll need to install `xz` first
+
+   ```zsh
+   brew86 install xz
+   pyenv install 3.10.6  # (or whichever version)
```

Comment:
🤔 hmm, I am not sure about this as a default suggestion. I think for folks who are doing work primarily with bq-etl or other repos that need rosetta, it could make a lot of sense & be simpler, but personally I need the arm64 stuff for most of my work (granted I am in a bit of a unique position). 

I *think* most of these tools being in rosetta isn't a problem, the way I had solved this problem was pyenv-alias, but I have had a bit of a running challenge with the sql formatter.